src/kernels/kernel.ts

```diff
@@ -70,8 +72,14 @@ import {
     NotebookCellRunState
 } from './types';
 import { KernelExecution } from '../notebooks/execution/kernelExecution';
+import { traceCellMessage } from '../notebooks/helpers';
+import { Cancellation } from '../client/common/cancellation';
 
 export class Kernel implements IKernel {
+    /**
+     * Used for debugging purposes, ability to uniquely identify kernels.
+     */
+    public readonly id: string;
```

Comment:
This was very useful, found we were creating a few kernels, initially I thought they were the same after adding this, found that we were actually creating new instances